### PR TITLE
eurocrops latvia

### PR DESCRIPTION
**Pull request by eddiechoi00**

### CODE REVIEW

File: fiboa_cli/datasets/ec_lv.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    'EC_hcat_n': 'EC_hcat_n', #fiboa custom field
+    'EC_hcat_c': 'EC_hcat_c' #fiboa custom field
+}
+
+def migration(gdf):
+    col = 'DATA_CHANG'
+    gdf[col] = pd.to_datetime(gdf[col], format = "%Y/%m/%d %H:%M:%S.%f", utc = True)
+    return gdf
+
+MIGRATION = migration
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_lv.py:60]:
```suggestion
COLUMN_MIGRATIONS = {'DATA_CHANG': lambda column: pd.to_datetime(column, format = "%Y/%m/%d %H:%M:%S.%f", utc = True)}
```

File: fiboa_cli/datasets/ec_lv.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    convert_(
+        output_file,
+        cache,
+        SOURCES,
+        COLUMNS,
+        ID,
+        TITLE,
+        DESCRIPTION,
+        input_files=input_files,
+        migration=MIGRATION,
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_lv.py:117]:
```suggestion
        column_migrations=COLUMN_MIGRATIONS,
```

**Comment by m-mohr** [comment]
Thanks. 

If I run the code I get:

> Column 'AREA_DECLARED' not found in dataset, removing from schema
> Column 'PERIOD_CODE' not found in dataset, removing from schema
> Column 'PRODUCT_CODE' not found in dataset, removing from schema

It looks like that needs from additional check what columns are available and how they are named.
Looks like the 10 character limit of shapefiles for columns hits here.

File: fiboa_cli/datasets/ec_lv.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+LICENSE = "CC-BY-4.0"
+
+COLUMNS = {
+    'geometry': 'geometry', #fiboa core field
+    'OBJECTID': 'id', #fiboa core field
+    'AREA_DECLARED': 'area', #fiboa core field
+    'DATA_CHANG': 'determination_datetime', #fiboa core field
+    'PERIOD_CODE': 'year', #fiboa custom field
+    'PARCEL_ID': 'parcel_id', #fiboa custom field
+    'PRODUCT_CODE': 'crop_id', #fiboa custom field
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_lv.py:46]:
```suggestion
    'AREA_DECLA': 'area', #fiboa core field
    'DATA_CHANG': 'determination_datetime', #fiboa core field
    'PERIOD_COD': 'year', #fiboa custom field
    'PARCEL_ID': 'parcel_id', #fiboa custom field
    'PRODUCT_CO': 'crop_id', #fiboa custom field
```

**Comment by m-mohr** [comment]
It looks like there might be encoding issues in the PRODUCT_DE column. Can you check that?

```
          PRODUCT_DE            EC_trans_n                       EC_hcat_n  \
0     Mie?i, vasaras         Barley summer                   spring_barley
1  Ilggad?gie z?l?ji  Permanent grasslands  pasture_meadow_grassland_grass
2     Rapsis, ziemas           Rape winter            winter_rapeseed_rape
3      Mie?i, ziemas         Barley winter                   winter_barley
4     Mie?i, vasaras         Barley summer                   spring_barley
```

**Comment by eddiechoi00** [comment]
My assumption seems to be right – I just tried translating a properly-encoded value (`Rapsis, ziemas`) and it comes out to "Rapeseed, winter" which seems to reflect `Rape winter` in `EC_trans_n`. I'm assuming the Latvian alphabet uses certain characters that had trouble with the encodings.

By making the code a comment, do you mean comment out the conversion of the column so the column doesn't get included in the final conversion for now?

**Comment by m-mohr** [comment]
Yes, exactly.

**Pull request closed without merging**